gateway-service/src/test/java/org/zowe/apiml/gateway/security/service/AuthenticationServiceTest.java

```diff
@@ -57,213 +58,277 @@
+            assertEquals("java.lang.String", jwtToken.getClass().getName());
+        }
 
-    @Test
-    void shouldValidateJwtToken() {
-        String jwtToken = authService.createJwtToken(USER, DOMAIN, LTPA);
+        @Test
+        void thenCreateValidJwtToken() {
+            when(jwtSecurityInitializer.getJwtPublicKey()).thenReturn(publicKey);
+            String jwtToken = authService.createJwtToken(USER, DOMAIN, LTPA);
 
-        TokenAuthentication token = new TokenAuthentication(jwtToken);
-        TokenAuthentication jwtValidation = authService.validateJwtToken(token);
+            TokenAuthentication token = new TokenAuthentication(jwtToken);
+            TokenAuthentication jwtValidation = authService.validateJwtToken(token);
 
-        assertEquals(USER, jwtValidation.getPrincipal());
-        assertEquals(jwtValidation.getCredentials(), jwtToken);
-        assertTrue(jwtValidation.isAuthenticated());
-    }
+            assertEquals(USER, jwtValidation.getPrincipal());
+            assertEquals(jwtValidation.getCredentials(), jwtToken);
+            assertTrue(jwtValidation.isAuthenticated());
+        }
 
-    @Test
-    void shouldThrowExceptionWhenTokenIsInvalid() {
-        String jwtToken = authService.createJwtToken(USER, DOMAIN, LTPA);
-        String brokenToken = jwtToken + "not";
-        TokenAuthentication token = new TokenAuthentication(brokenToken);
-        assertThrows(
-            TokenNotValidException.class,
-            () -> authService.validateJwtToken(token)
-        );
-    }
+        @Test
+        void thenParseJwtTokenAsQueryResponse() {
+            String jwtToken = authService.createJwtToken(USER, DOMAIN, LTPA);
 
-    @Test
-    void shouldThrowExceptionWhenTokenIsExpired() {
-        TokenAuthentication token = new TokenAuthentication(createExpiredJwtToken(privateKey));
-        assertThrows(
-            TokenExpireException.class,
-            () -> authService.validateJwtToken(token)
-        );
-    }
+            String dateNow = new Date().toString().substring(0, 16);
+            QueryResponse parsedToken = authService.parseJwtToken(jwtToken);
 
+            assertEquals(QueryResponse.class, parsedToken.getClass());
+            assertEquals(DOMAIN, parsedToken.getDomain());
+            assertEquals(USER, parsedToken.getUserId());
+            assertEquals(parsedToken.getCreation().toString().substring(0, 16), dateNow);
+            Date toBeExpired = DateUtils.addHours(parsedToken.getCreation(), 8);
+            assertEquals(parsedToken.getExpiration(), toBeExpired);
+        }
 
-    @Test
-    void shouldThrowExceptionWhenOccurUnexpectedException() {
-        assertThrows(
-            TokenNotValidException.class,
-            () -> authService.validateJwtToken((String) null)
-        );
-    }
+        @Test
+        void thenGetTokenWithDefaultExpiration() {
+            String jwt1 = authService.createJwtToken("user", "domain", "ltpaToken");
```

Comment:
Can the variables be used instead of string literals? `LTPA` looks like it can be used at least.